File: docs/components_page/components/input/radio_check.R
library(dashBootstrapComponents)
library(dashHtmlComponents)

radioitems <- htmlDiv(
  list(
    dbcLabel("Choose one"),
    dbcRadioItems(
      options = list(
        list(label = "Option 1", value = 1),
        list(label = "Option 2", value = 2),
        list(label = "Disabled Option", value = 3, disabled = TRUE)
      ),
      value = 1,
      id = "radioitems-input"
    )
  )
)


checklist <- htmlDiv(
  list(
    dbcLabel("Choose a bunch"),
    dbcChecklist(
      options = list(
        list(label = "Option 1", value = 1),
        list(label = "Option 2", value = 2),
        list(label = "Disabled Option", value = 3, disabled = TRUE)
      ),
      value = list(1),
      id = "checklist-input"
    )
  )
)

switches <- htmlDiv(
  list(
    dbcLabel("Toggle a bunch"),
    dbcChecklist(
      options = list(
        list(label = "Option 1", value = 1),
        list(label = "Option 2", value = 2),
        list(label = "Disabled Option", value = 3, disabled = TRUE)
      ),
      value = list(1),
      id = "switches-input",
      switch = TRUE,
    )
  )
)

inputs <- htmlDiv(
  list(
    dbcForm(list(radioitems, checklist, switches)),
    htmlP(id = "radioitems-checklist-output")
  )
)


app$callback(
  output("radioitems-checklist-output", "children"),
  list(
    input("radioitems-input", "value"),
    input("checklist-input", "value"),
    input("switches-input", "value")
  ),
  function(radio_items_value, checklist_value, switches_value) {
    n_checkboxes <- length(checklist_value)
    n_switches <- length(switches_value)
    s <- if (n_checkboxes != 1) "s" else ""
    es <- if (n_switches != 1) "es" else ""

    template <- "Radio button %d,  %d checklist item%s and %d switch%s selected."

    return(
      sprintf(template, radio_items_value, n_checkboxes, s, n_switches, es)
    )
  }
)

File: docs/components_page/components/input/radio_check.jl
using DashBootstrapComponents, DashHtmlComponents

radioitems = html_div([
    dbc_label("Choose one"),
    dbc_radioitems(
        options = [
            Dict("label" => "Option 1", "value" => 1),
            Dict("label" => "Option 2", "value" => 2),
            Dict("label" => "Disabled Option", "value" => 3, "disabled" => true),
        ],
        value = 1,
        id = "radioitems-input",
    ),
]);

checklist = html_div([
    dbc_label("Choose a bunch"),
    dbc_checklist(
        options = [
            Dict("label" => "Option 1", "value" => 1),
            Dict("label" => "Option 2", "value" => 2),
            Dict("label" => "Disabled Option", "value" => 3, "disabled" => true),
        ],
        value = [1],
        id = "checklist-input",
    ),
]);

switches = html_div([
    dbc_label("Toggle a bunch"),
    dbc_checklist(
        options = [
            Dict("label" => "Option 1", "value" => 1),
            Dict("label" => "Option 2", "value" => 2),
            Dict("label" => "Disabled Option", "value" => 3, "disabled" => true),
        ],
        value = [1],
        id = "switches-input",
        switch = true,
    ),
]);

inputs = html_div([
    dbc_form([radioitems, checklist, switches]),
    html_p(id = "radioitems-checklist-output"),
]);


callback!(
    app,
    Output("radioitems-checklist-output", "children"),
    Input("radioitems-input", "value"),
    Input("checklist-input", "value"),
    Input("switches-input", "value"),
) do radio_items_value, checklist_value, switches_value

    n_checkboxes = length(checklist_value)
    n_switches = length(switches_value)

    output_string =
        """Radio button $radio_items_value, $n_checkboxes """ *
        """checklist item$(n_checkboxes != 1 ? "s" : "") and $n_switches """ *
        """switch$(n_switches != 1 ? "es" : "") selected."""

    return output_string
end;
